tests/testutil/service.go

```diff
@@ -202,6 +203,76 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+func (ts *TestService) GetEndpointUrl() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+func (ts *TestService) GetKubeEndpointUrlByRevision(revision string) string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	args := append([]string{"--format", "json",
+		fmt.Sprintf("%s:%s", strings.Split(ts.Name, ":")[0], revision)})
+	out, err := RioCmd("inspect", args)
+	ts.Service = riov1.Service{}
+	if err := json.Unmarshal([]byte(out), &ts.Service); err != nil {
+		return ""
+	}
+	args = append([]string{"get", "service.rio.cattle.io",
```

Comment:
If we reuse this over and over again, we should put it in a const by `testingNamespace`, but we can leave for now.